### PR TITLE
Fix index delete - again!

### PR DESCRIPTION
**Pull request by logan-markewich**
One more delete bug!

If you construct a vector index using the documents from the client, calling delete will never work if we raise a ValueError

```python
index = GPTQdrantIndex([], client=qdrant_client_instance, collection_name=self._collection_name)
index.delete("some_remote_id")
```

### CODE REVIEW

File: gpt_index/data_structs/data_structs_v2.py [hunk 1]
```diff
@@ -201,9 +200,10 @@ def add_node(
     def delete(self, doc_id: str) -> None:
         """Delete a Node."""
         if doc_id not in self.doc_id_dict:
-            raise ValueError("doc_id not found in doc_id_dict")
```
Comment (jerryjliu) [inline gpt_index/data_structs/data_structs_v2.py:204]: out of curiosity, does this primarily apply to vector indexes that aren't the simple vector index?

Comment (logan-markewich) [reply gpt_index/data_structs/data_structs_v2.py:204]: @jerryjliu pretty much! But I didn't see the harm in applying it to all vector indexes 🤔

Comment (logan-markewich) [reply gpt_index/data_structs/data_structs_v2.py:204]: Is there a better option you were thinking of? With the 3rd party vector stores, the documents are sometimes in the remote client (which we call delete for after the index struct), and sometimes also in the index, so we still want to delete it if it's there

Comment (jerryjliu) [reply gpt_index/data_structs/data_structs_v2.py:204]: yeah, that makes sense to me. there is a bit of messiness in deciding what data goes into vector store and what doesn't, will take a look at cleaning it up

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**